metals/src/main/scala/scala/meta/internal/metals/scalacli/ScalaCli.scala

```diff
@@ -231,60 +239,59 @@ class ScalaCli(
   }
 
   def loaded(path: AbsolutePath): Boolean =
-    roots0.contains(path)
-
-  def start(roots: Seq[AbsolutePath]): Future[Unit] = {
+    ifConnectedOrElse(st =>
+      st.path == path || path.toNIO.startsWith(st.path.toNIO)
+    )(false)
 
+  def start(path: AbsolutePath): Future[Unit] = {
     disconnectOldBuildServer().onComplete {
       case Failure(e) =>
         scribe.warn("Error disconnecting old Scala CLI server", e)
       case Success(()) =>
     }
 
-    val command =
-      baseCommand ++ Seq(roots.head.toString) ++ roots.tail.flatMap(p =>
-        Seq(";", p.toString)
+    val command = baseCommand :+ path.toString()
+
+    val connDir = if (path.isDirectory) path else path.parent
+
+    val nextSt = ConnectionState.Connecting(path)
+    if (state.compareAndSet(ConnectionState.Empty, nextSt)) {
+      val futureConn = BuildServerConnection.fromSockets(
+        connDir,
+        buildClient(),
+        languageClient,
+        () => ScalaCli.socketConn(command, connDir),
+        tables().dismissedNotifications.ReconnectScalaCli,
+        config(),
+        "Scala CLI",
+        supportsWrappedSources = Some(true),
       )
 
-    val futureConn = BuildServerConnection.fromSockets(
-      workspace(),
-      buildClient(),
-      languageClient,
-      () => ScalaCli.socketConn(command, workspace()),
-      tables().dismissedNotifications.ReconnectScalaCli,
-      config(),
-      "Scala CLI",
-      supportsWrappedSources = Some(true),
-    )
-
-    val f = futureConn.flatMap(connectToNewBuildServer(_, roots))
-
-    f.transform {
-      case Failure(ex) =>
-        scribe.error("Error starting Scala CLI", ex)
-        Success(())
-      case Success(_) =>
-        scribe.info("Scala CLI started")
-        Success(())
+      val f = futureConn.flatMap { conn =>
+        state.set(ConnectionState.Connected(path, conn, ImportedBuild.empty))
+        scribe.info(s"Connected to Scala CLI server v${conn.version}")
+        importBuild().map { _ =>
+          BuildChange.Reconnected
+        }
+      }
+
+      f.transform {
+        case Failure(ex) =>
+          scribe.error("Error starting Scala CLI", ex)
+          Success(())
+        case Success(_) =>
+          scribe.info("Scala CLI started")
+          Success(())
+      }
+    } else {
+      scribe.error(s"Multiply Scala CLI start calls. Failed for: $path")
```

Comment:
It sounds like, now we can't import multiple different scripts/directories to ScalaCLI at the same time, right?
It would be an issue if users create multiple scala cli scripts in source directories that belong to the base project.

I think it's ok for now, but it would be awesome if we can support importing multiple scripts/directories.

Reply:
> I think it's ok for now, but it would be awesome if we can support importing multiple scripts/directories.

yep, I was thinking to make support for multiply but it would take much more time.

Also, in general it's really close to multiply-bsp servers /multi-workspace support. I would implement it at first and then it will be easy to apply for this particular scala-cli case

Reply:
sounds good, hope users don't create many sc files in source directories 😄